### PR TITLE
qa/tasks/cephadm: update_info_yam() only if ctx.archive is valid

### PR DESCRIPTION
**Pull request by ideepika**
related to: 6cd8e4b55883b2d282f101f11521a2759b3db564
Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: qa/tasks/cephadm.py [hunk 1]
```diff
@@ -155,20 +155,26 @@ def download_cephadm(ctx, config, ref):
             )
 
 @contextlib.contextmanager
-def ceph_log(ctx, config):
-    cluster_name = config['cluster']
-    fsid = ctx.ceph[cluster_name].fsid
-
-    # Add logs directory to job's info log file
-    with open(os.path.join(ctx.archive, 'info.yaml'), 'r+') as info_file:
+def update_archive_setting(archive_dir, key, value):
```
Comment (tchaikov) [inline qa/tasks/cephadm.py:158]: why is this method decorated using `contextlib.contextmanager`?

**Review by tchaikov — COMMENTED** [review]
please split this change into two commit

1. refactor
1. `update_archive_setting() only if ctx.archive is valid`

File: qa/tasks/cephadm.py (hunk 1 above)
Comment (sebastian-philipp) [reply qa/tasks/cephadm.py:158]: Also `update_archive_setting` seems to be a strange name, as this function actually **doesn't update the archive setting**.

Can you find a better name?

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

File: qa/tasks/cephadm.py (hunk 1 above)
Comment (ideepika) [reply qa/tasks/cephadm.py]: @tchaikov made updates, thanks for feedback!
@sebastian-philipp what about `s/update_archive_setting/update_info_yaml` ?

**Comment by sebastian-philipp** [comment]
https://pulpito.ceph.com/swagner-2020-12-09_08:09:28-rados:cephadm-wip-swagner-testing-2020-12-08-1410-distro-basic-smithi/

looks reasonable, but we have many failures:

* docker rate limit on ubuntu+podman
* https://tracker.ceph.com/issues/48142
* https://tracker.ceph.com/issues/48535
* dashboard e2e test

**Review by sebastian-philipp — APPROVED** [review]

**Pull request merged**